### PR TITLE
3 switching discussions

### PR DESCRIPTION
**Pull request by koleksiuk**

### CODE REVIEW

File: app/stores/ThreadStore.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+var Reflux = require('reflux');
+var ThreadActions = require('../actions/ThreadActions.jsx');
+var Message = require('../models/Message.jsx');
+var moment = require("moment") // remove when default is removed
+
+var defaultMessages = [
+  new Message({ userId: 1, discussionId: 1, body: "Cześć", timestamp: Number(moment().subtract(2, "minutes"))  }),
```
Comment (koleksiuk) [inline app/stores/ThreadStore.jsx]: don't keep discussionId in messages, keep it in store state.

File: app/stores/ThreadStore.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  new Message({ userId: 2, discussionId: 1, body: "Cześć", timestamp: Number(moment().subtract(40, "seconds"))}),
+  new Message({ userId: 1, discussionId: 1, body: "Co tam?", timestamp: Number(moment().subtract(10, "seconds"))})
+]
+
+var ThreadStore = Reflux.createStore({
+  listenables: [ThreadActions],
```
Comment (koleksiuk) [inline app/stores/ThreadStore.jsx]: [ThreadActions, ChatActions]

onSwitchDiscussion: function...

File: app/dao/MessagesDAO.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    this.userId = userId;
+  }
+
+  setDiscussionId(discussionId) {
+    this.discussionId = discussionId;
+
+    return this;
+  }
+
+  create(userId, body) {
```
Comment (Bajena) [inline app/dao/MessagesDAO.js]: userId is unused here, are we going to implement this later?

File: app/dao/UsersDAO.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+var User = require("../models/User.js");
+
+var USERS = {
+  "1": new User({ id: 1, name: "Janek" }),
+  "2": new User({ id: 2, name: "Konrad" })
+};
+
+class UsersDAO {
```
Comment (Bajena) [inline app/dao/UsersDAO.js]: This actually is a repository rather than a DAO, but it's fine ;)

File: app/models/Message.js [hunk 5]
```diff
@@ -17,6 +17,10 @@ class Message {
     this.id = id;
   }
 
+  getDiscussion() {
```
Comment (Bajena) [inline app/models/Message.js]: This method is probably unnecessary for now?

File: app/dao/MessagesDAO.js (hunk 3 above)
Comment (koleksiuk) [inline app/dao/MessagesDAO.js]: maybe it should come from session?

File: app/stores/ChatStore.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+var DISCUSSIONS = [
+  new Discussion({ id: 1, name: "Team channel", userIds: [1, 2]}),
+  new Discussion({ id: 2, name: "Fellow devs" , userIds: [1, 2]})
+];
+
+var ChatStore = Reflux.createStore({
+  listenables: [ChatActions],
+
+  onSwitchDiscussion: function(discussion) {
+    this.currentDiscussion = discussion;
```
Comment (Bajena) [inline app/stores/ChatStore.js]: MessagesDAO currentDiscussion should be updated from here probably

File: app/stores/ChatStore.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  },
+
+  getDiscussions: function() {
```
Comment (Bajena) [inline app/stores/ChatStore.js]: Is it needed anymore?

Comment (koleksiuk) [inline app/stores/ChatStore.js]: not :P

**Pull request merged**